src/Raven.Server/RavenServer.cs

```diff
@@ -1979,7 +1979,7 @@ private void ListenToNewTcpConnection(TcpListener listener)
                         _tcpLogger.Info("Failure when processing tcp connection", e);
                     }
                 }
-            });
+            }, TaskCreationOptions.LongRunning);
```

Comment:
Why have this? In most cases, we will shift to our own thread. Even for something like Elector, they are commonly relatively short. Here you are forcing new thread creation on each connection

Reply:
Reverted. I've added that when Elector was sync.